apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.js

```diff
@@ -81,256 +95,183 @@ router.post(
   }),
 );
 
-router.post('/', function (req, res, next) {
-  if (res.locals.question.course_id !== res.locals.course.id) {
-    return next(error.make(403, 'Access denied'));
-  }
-  if (req.body.__action === 'change_id') {
-    debug(`Change qid from ${res.locals.question.qid} to ${req.body.id}`);
-    if (!req.body.id) return next(new Error(`Invalid QID (was falsy): ${req.body.id}`));
-    if (!/^[-A-Za-z0-9_/]+$/.test(req.body.id)) {
-      return next(
-        new Error(
-          `Invalid QID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.id}`,
-        ),
-      );
-    }
-    let qid_new;
-    try {
-      qid_new = path.normalize(req.body.id);
-    } catch (err) {
-      return next(new Error(`Invalid QID (could not be normalized): ${req.body.id}`));
+router.post(
+  '/',
+  asyncHandler(async (req, res, next) => {
+    if (res.locals.question.course_id !== res.locals.course.id) {
+      throw error.make(403, 'Access denied');
     }
-    if (res.locals.question.qid === qid_new) {
-      debug('The new qid is the same as the old qid - do nothing');
-      res.redirect(req.originalUrl);
-    } else {
-      const editor = new QuestionRenameEditor({
-        locals: res.locals,
-        qid_new: qid_new,
-      });
-      editor.canEdit((err) => {
-        if (ERR(err, next)) return;
-        editor.doEdit((err, job_sequence_id) => {
-          if (ERR(err, (e) => logger.error('Error in doEdit()', e))) {
-            res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
-          } else {
-            res.redirect(req.originalUrl);
-          }
+    if (req.body.__action === 'change_id') {
+      if (!req.body.id) throw new Error(`Invalid QID (was falsy): ${req.body.id}`);
```

Comment:
As long as we're changing all these: let's specify a 4XX error code (`error.make(4..)`). I think 400 would be appropriate here and elsewhere below where we do similar validation.

Reply:
Updated these to throw a 400 error code in [15e6a55](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/15e6a551ace1b217ed76fec735c5306f682080d1).